src/tests/ftest/util/macsio_test_base.py

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/python
+"""
+  (C) Copyright 2020 Intel Corporation.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+  GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+  The Government's rights to use, modify, reproduce, release, perform, display,
+  or disclose this software are subject to the terms of the Apache License as
+  provided in Contract No. B609815.
+  Any reproduction of computer software, computer software documentation, or
+  portions thereof marked with this legend must also reproduce the markings.
+"""
+from apricot import TestWithServers
+from command_utils_base import CommandFailure
+from job_manager_utils import Mpirun
+from macsio_util import MacsioCommand
+
+
+class MacsioTestBase(TestWithServers):
+    """Base MACSio test class.
+
+    :avocado: recursive
+    """
+
+    def __init__(self, *args, **kwargs):
+        """Initialize a MacsioTestBase object."""
+        super(MacsioTestBase, self).__init__(*args, **kwargs)
+        self.manager = None
+        self.macsio = None
+
+    def setUp(self):
+        """Set up each test case."""
+        super(MacsioTestBase, self).setUp()
+        self.manager = Mpirun(None, subprocess=False, mpitype="mpich")
```

Comment:
Do you need to consider the MPI env when hdf5 is built with openmpi?   Does Mpirun() use a full path ?  The concern here is what happens to LD_LIBRARY_PATH when you use Mpirun()?    If nothing then you should be ok to launch Macsio with mpich even when hdf5 is built with openmpi.

Reply:
The `Mpirun` class handles setting the LD_LIBRARY_PATH with https://github.com/daos-stack/daos/blob/master/src/tests/ftest/util/job_manager_utils.py#L346